pingcap/tidb-operator
# Add readinessProbe config for tidb

### What problem does this PR solve? 
Add readinessProbe config for tidb to support probe tidb's readiness by requesting 127.0.0.0:10080/status

fix #2132 
### What is changed and how does it work?
ref https://github.com/pingcap/tidb/pull/20649
Add this config support like this:
```
tidb:
  readinessProbe:
    type: command
```
This will make tidb pod use Exec handle to run curl probe the url 127.0.0.1:10080/status
Default still use TCPPPort(4000) for compatibility
### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
    * enable TLS following https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/enable-tls-between-components
    * deploy cluster with readinessProbe.status.API setted
    * check tidb can be ready


Side effects


Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Add `readinessProbe` config for tidb to support probe tidb's readiness by requesting 127.0.0.0:10080/status
```
